src/Simulator/IncrementalRedstoneSimulator/RedstoneHandler.cpp

```diff
@@ -30,59 +30,56 @@
+		case E_BLOCK_ACACIA_FENCE_GATE:                                                 \
+		case E_BLOCK_BIRCH_FENCE_GATE:                                                  \
+		case E_BLOCK_DARK_OAK_FENCE_GATE:                                               \
+		case E_BLOCK_FENCE_GATE:                                                        \
+		case E_BLOCK_IRON_TRAPDOOR:                                                     \
+		case E_BLOCK_JUNGLE_FENCE_GATE:                                                 \
+		case E_BLOCK_SPRUCE_FENCE_GATE:                                                 \
+		case E_BLOCK_TRAPDOOR:              return SmallGateHandler::Callback;          \
+		case E_BLOCK_REDSTONE_LAMP_OFF:                                                 \
+		case E_BLOCK_REDSTONE_LAMP_ON:      return RedstoneLampHandler::Callback;       \
+		case E_BLOCK_REDSTONE_REPEATER_OFF:                                             \
+		case E_BLOCK_REDSTONE_REPEATER_ON:  return RedstoneRepeaterHandler::Callback;   \
+		case E_BLOCK_REDSTONE_TORCH_OFF:                                                \
+		case E_BLOCK_REDSTONE_TORCH_ON:     return RedstoneTorchHandler::Callback;      \
+		case E_BLOCK_OBSERVER:              return ObserverHandler::Callback;           \
+		case E_BLOCK_PISTON:                                                            \
+		case E_BLOCK_STICKY_PISTON:         return PistonHandler::Callback;             \
+		case E_BLOCK_LEVER:                                                             \
+		case E_BLOCK_STONE_BUTTON:                                                      \
+		case E_BLOCK_WOODEN_BUTTON:         return RedstoneToggleHandler::Callback;     \
+		case E_BLOCK_BLOCK_OF_REDSTONE:     return RedstoneBlockHandler::Callback;      \
+		case E_BLOCK_COMMAND_BLOCK:         return CommandBlockHandler::Callback;       \
+		case E_BLOCK_HOPPER:                return HopperHandler::Callback;             \
+		case E_BLOCK_NOTE_BLOCK:            return NoteBlockHandler::Callback;          \
+		case E_BLOCK_REDSTONE_WIRE:         return RedstoneWireHandler::Callback;       \
+		case E_BLOCK_TNT:                   return TNTHandler::Callback;                \
+		case E_BLOCK_TRAPPED_CHEST:         return TrappedChestHandler::Callback;       \
+		case E_BLOCK_TRIPWIRE_HOOK:         return TripwireHookHandler::Callback;       \
+		default:                                                                        \
+		{                                                                               \
+			if (cBlockDoorHandler::IsDoorBlockType(BlockType))                          \
+			{                                                                           \
+				return DoorHandler::Callback;                                           \
+			}                                                                           \
+		}                                                                               \
+	}
```

Comment:
You say that "Everyone agrees the `do {} while (false)` thing is weird and confusing" but clearly not. If I thought it was bad, I wouldn't have done it.

It makes macros behave syntactically more like functions which can only be a good thing.

Reply:
For the uninitiated, do while false reads like dead code which should be removed, and they would be right and everything would compile and work fine. Until they hit a clang-tidy error, of all things, complaining about a semicolon, before realising that it is there not to do anything, but to satisfy a style checker. If we claim to be stewards of maintainable and _obvious_ code then we can't be introducing these C-isms.

On the other hand it's true this is being used already, so consistency dictates we'd do the same here. :/